lib/react_client/react_interop.dart

```diff
@@ -34,8 +36,54 @@ abstract class React {
 
   external static bool isValidElement(dynamic object);
   external static ReactClass get Fragment;
+
+  /// Returns a ref that can be attached to an element via the ref attribute.
+  ///
+  /// When a ref is passed to an element in render, a reference to the node becomes
+  /// accessible at the 'current' attribute of the ref.
+  ///
+  /// See: <https://reactjs.org/docs/refs-and-the-dom.html#creating-refs>.
+  external static createRef();
 }
 
+class createRef<CurrentType> {
```

Comment:
This class and its members need doc comments.

The class comment should also have an example usage, showcasing the generic typing, and setting a convention for usage.

Something like
```dart
class FooComponent extends react.Component2 {
  final Ref<BarComponent> barRef = createRef();
  final Ref<Element> inputRef = createRef();

  render() => [
    Bar({'ref': barRef}),
    react.input({'ref': inputRef}),
  ];
}
```